### PR TITLE
Hook up list of accounts page

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Display actual list of accounts and editing of account names.

## Changes

- Hook up Accounts page to actual data
- Allow renaming accounts
- Make types of `useWritableSelectedAccount` more precise.

### CODE REVIEW

File: packages/browser-wallet/src/popup/popupX/pages/Accounts/Accounts.tsx [hunk 1]
```diff
@@ -13,34 +15,149 @@ import Text from '@popup/popupX/shared/Text';
...
+                        value={editedName}
+                        placeholder={fallbackName}
+                        onChange={onInputChange}
+                        onKeyUp={onKeyUp}
+                        maxLength={25}
+                    />
+                </Text.Main>
+                <Button.Icon className="transparent" icon={<Checkmark />} onClick={onComplete} />
+                <Button.Icon className="transparent" icon={<Close />} onClick={onAbort} />
+            </>
```
Comment (soerenbf) [inline packages/browser-wallet/src/popup/popupX/pages/Accounts/Accounts.tsx:77]: I would suggest something like
```suggestion
            <>
                <Text.Main>
                    <input
                        autoFocus
                        className="editable"
                        value={editedName}
                        placeholder={fallbackName}
                        onChange={onInputChange}
                        onKeyUp={onKeyUp}
                        maxLength={25}
                    />
                </Text.Main>
                <div className="flex-row">
                    <Button.Icon className="transparent m-r-20" icon={<Checkmark className="checkmark-icon" />} onClick={onComplete} />
                    <Button.Icon className="transparent" icon={<Close />} onClick={onAbort} />
                </div
            </>
```

File: packages/browser-wallet/src/popup/popupX/pages/Accounts/Accounts.scss [hunk 2]
```diff
@@ -25,9 +25,32 @@
...
+                outline: none;
+            }
+        }
+
         .button__icon {
             gap: rem(16px);
 
+            svg {
+                width: rem(16px);
+            }
```
Comment (soerenbf) [inline packages/browser-wallet/src/popup/popupX/pages/Accounts/Accounts.scss:52]:
```suggestion
            svg {
                width: rem(16px);
                
                &.checkmark-icon {
                    width: rem(12px);
                }
            }
```

Assuming the corresponding class is added to the icon in the view

**Review by soerenbf — COMMENTED** [review]
A couple of ideas

File: packages/browser-wallet/src/popup/popupX/pages/Accounts/Accounts.tsx [hunk 3]
```diff
@@ -13,34 +15,149 @@ import Text from '@popup/popupX/shared/Text';
...
+function AccountListItem({ credential }: AccountListItemProps) {
     const { t } = useTranslation('x', { keyPrefix: 'accounts' });
     const nav = useNavigate();
     const navToPrivateKey = () => nav(absoluteRoutes.settings.accounts.privateKey.path);
     const navToConnectedSites = () => nav(absoluteRoutes.settings.accounts.connectedSites.path);
     const navToIdCards = () => nav(absoluteRoutes.settings.idCards.path);
+    const identityName = useIdentityName(credential);
+    const accountInfo = useAccountInfo(credential);
+    const setAccount = useWritableSelectedAccount(credential.address);
+    const fallbackName = fallbackAccountName(credential.credNumber);
```
[the remark this answers is not in the document]
Comment (Ivan-Mahda) [reply packages/browser-wallet/src/popup/popupX/pages/Accounts/Accounts.tsx:101]: My suggestion, default name for account should be SplitAddress, similar to current implementation. Until the user wants to change the name. And only then we show this account name.
For me, these Account 1,2,3 are just placeholders in the design

[the remark this answers is not in the document]
Comment (limemloh) [reply packages/browser-wallet/src/popup/popupX/pages/Accounts/Accounts.tsx]: I have updated this, please confirm that this is what you meant :smile:

File: packages/browser-wallet/src/popup/popupX/pages/Accounts/Accounts.tsx (hunk 1 above)
Comment (limemloh) [reply packages/browser-wallet/src/popup/popupX/pages/Accounts/Accounts.tsx]: I have done something like this now

**Review by Ivan-Mahda — APPROVED** [review]

File: packages/browser-wallet/src/popup/popupX/pages/Accounts/Accounts.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (soerenbf) [reply packages/browser-wallet/src/popup/popupX/pages/Accounts/Accounts.tsx]: Looks good 😄

**Review by soerenbf — APPROVED** [review]

**Pull request merged**